samples/csharp/common/AutoML/ConsoleHelper.cs

```diff
@@ -89,6 +98,11 @@ internal static void PrintIterationMetrics(int iteration, string trainerName, Re
             CreateRow($"{iteration,-4} {trainerName,-35} {metrics?.RSquared ?? double.NaN,8:F4} {metrics?.MeanAbsoluteError ?? double.NaN,13:F2} {metrics?.MeanSquaredError ?? double.NaN,12:F2} {metrics?.RootMeanSquaredError ?? double.NaN,8:F2} {runtimeInSeconds.Value,9:F1}", Width);
         }
 
+        internal static void PrintIterationMetrics(int iteration, string trainerName, RankingMetrics metrics, double? runtimeInSeconds)
+        {
+            CreateRow($"{iteration,-4} {trainerName,-35} {metrics?.NormalizedDiscountedCumulativeGains[0] ?? double.NaN,9:F4} {metrics?.NormalizedDiscountedCumulativeGains[2] ?? double.NaN,9:F4} {metrics?.NormalizedDiscountedCumulativeGains[9] ?? double.NaN,9:F4}  {metrics?.DiscountedCumulativeGains[9] ?? double.NaN,9:F4} {runtimeInSeconds.Value,9:F1}", Width);
```

Comment:
What is the length of the array? This will be max `k` in `NDCG@k`, and likely also the default set in the AutoML code for `k` for which NDCG value to optimize (the optimization metric).

Is there an input for `k` in the AutoML API? If so, it would be good to set to perhaps `10`, and ensure the user's specified `k` value is within the printed metrics. 

Setting a value in a sample has a good pedagogical value as it lets users know they should be setting this value. The input would control which `k` in `NDCG@k` that AutoML should optimize towards (aka, the optimization metric).

Reply:
Unfortunately, for the progress handler, I'm unable to change the number of `k`. For the `Evaluate` method, however, I can specify it.

Reply:
Yes, you're right; it's not exposed. I left a comment on the lack of the `k` parameter for the AutoML ranking experiment settings. 

https://github.com/dotnet/machinelearning-samples/pull/852#discussion_r498425201
> As an aside the AutoML code isn't passing through a `k` parameter for the optimization metric. This is room for improvement in our AutoML ranking API/code. Currently the AutoML code is hard-coded to optimize NDCG@10 (or DCG@10) instead of a user supplied `k` in `NDCG@k`. Having the user supply this parameter instead of hard coding lets the AutoML code optimize the model towards the user's needs; for instance NDCG@3 is useful for an auto-suggest showing three results, NDCG@10 is useful for the first page of search results, NDCG@30 optimizes the first three pages.
>
> TLDR; an `@k` parameter should be exposed in the AutoML ranking experiment settings.

#RESOLVED